### PR TITLE
Actions with clone() pattern 1/3

### PR DESCRIPTION
**Pull request by ricardoquesada**
This is the 1st patch of 3.

1)
Removes the copyWithZone() patterns.
Adds the clone() pattern instead (which is more C++ friendly)
Adds also "const" to getters.

### CODE REVIEW

File: cocos2dx/actions/CCAction.cpp [hunk 1]
```diff
@@ -37,12 +37,23 @@ of this software and associated documentation files (the "Software"), to deal
 //
 
 CCAction::CCAction()
-:m_pOriginalTarget(NULL)
+:CCObject()
```
Comment (minggo) [inline cocos2dx/actions/CCAction.cpp]: Why should invoke father's constructor explicitly?

Comment (ricardoquesada) [inline cocos2dx/actions/CCAction.cpp]: yep... my bad, there is no need to call it explicitly since it is the default constructor.

**Comment by ricardoquesada** [comment]
@minggo 
fixed.

**Comment by dumganhar** [comment]
Hi Riq, 
I merged my PR in bindings-generator(https://github.com/cocos2d/bindings-generator/pull/9).
It will not generate jsbinding glue codes for copy constructor.

**Pull request merged**